frontend/src/metabase/query_builder/components/view/sidebars/ChartSettingsSidebar.jsx

```diff
@@ -5,38 +5,49 @@ import ChartSettings from "metabase/visualizations/components/ChartSettings";
 import visualizations from "metabase/visualizations";
 import SidebarContent from "metabase/query_builder/components/SidebarContent";
 
-const ChartSettingsSidebar = ({
-  question,
-  result,
-  addField,
-  initialChartSetting,
-  onReplaceAllVisualizationSettings,
-  onClose,
-  onOpenChartType,
-  ...props
-}) =>
-  result && (
-    <SidebarContent
-      className="full-height"
-      title={t`${visualizations.get(question.display()).uiName} options`}
-      onDone={onClose}
-      onBack={onOpenChartType}
-    >
-      <ChartSettings
-        question={question}
-        addField={addField}
-        series={[
-          {
-            card: question.card(),
-            data: result.data,
-          },
-        ]}
-        onChange={onReplaceAllVisualizationSettings}
-        onClose={onClose}
-        noPreview
-        initial={initialChartSetting}
-      />
-    </SidebarContent>
-  );
+export default class ChartSettingsSidebar extends React.Component {
+  state = { sidebarPropsOverride: undefined };
```

Comment:
FYI typically I'd use `null` here, but I can't really articulate why at the moment :)

Reply:
yeah that's fair. I guess it's kinda weird to _define_ something as `undefined`.